lib/gnat.ex

```diff
@@ -141,10 +135,10 @@ defmodule Gnat do
     sub = Command.build(:sub, request.inbox, sid, [])
     unsub = Command.build(:unsub, sid, [max_messages: 1])
     pub = Command.build(:pub, request.topic, request.body, reply_to: request.inbox)
-    receivers = Map.put(state.receivers, sid, request.recipient)
-    next_sid = sid + 1
     :ok = :gen_tcp.send(state.tcp, [sub, unsub, pub])
-    {:reply, {:ok, sid}, %{state | receivers: receivers, next_sid: next_sid}}
+    state = add_subscription_to_state(state, sid, request.recipient) |> cleanup_subscription_from_state(sid, max_messages: 1)
+    next_sid = sid + 1
+    {:reply, {:ok, sid}, %{state | next_sid: next_sid}}
   end
   def handle_call({:unsub, sid, opts}, _from, state) do
```

Comment:
While you're here. You should make it so this checks your internal state before sending an unsub command.